shell/platform/common/client_wrapper/core_implementations.cc

```diff
@@ -36,17 +36,31 @@ void ForwardToHandler(FlutterDesktopMessengerRef messenger,
                       const FlutterDesktopMessage* message,
                       void* user_data) {
   auto* response_handle = message->response_handle;
-  BinaryReply reply_handler = [messenger, response_handle](
+  auto messenger_ptr = std::shared_ptr<FlutterDesktopMessenger>(
+      FlutterDesktopMessengerAddRef(messenger),
+      &FlutterDesktopMessengerRelease);
+  BinaryReply reply_handler = [messenger_ptr, response_handle](
                                   const uint8_t* reply,
                                   size_t reply_size) mutable {
+    // Note: This can be called on any thread.
+    auto lock = std::unique_ptr<FlutterDesktopMessenger,
```

Comment:
I think it would help readability to typedef this. Maybe `StackBasedMessengerLock`?

Reply:
Done, `FlutterDesktopMessengerScopedLock`